### PR TITLE
fix(xds): explicitly set initial fetch timeout to zero to keep Envoy wait for xds resources

### PR DESCRIPTION
**Pull request by jijiechen**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues 
  - fixes https://github.com/kumahq/kuma/issues/11023
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - Confirmed
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) 
  - Tested
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? 
  - No
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
  - yes

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/xds/bootstrap/template_v3.go [hunk 1]
```diff
@@ -158,10 +159,12 @@ func genConfig(parameters configParameters, proxyConfig xds.Proxy, enableReloada
 		DynamicResources: &envoy_bootstrap_v3.Bootstrap_DynamicResources{
 			LdsConfig: &envoy_core_v3.ConfigSource{
 				ConfigSourceSpecifier: &envoy_core_v3.ConfigSource_Ads{Ads: &envoy_core_v3.AggregatedConfigSource{}},
+				InitialFetchTimeout:   durationpb.New(0),
 				ResourceApiVersion:    envoy_core_v3.ApiVersion_V3,
 			},
 			CdsConfig: &envoy_core_v3.ConfigSource{
 				ConfigSourceSpecifier: &envoy_core_v3.ConfigSource_Ads{Ads: &envoy_core_v3.AggregatedConfigSource{}},
+				InitialFetchTimeout:   durationpb.New(0),
```
Comment (michaelbeaumont) [inline pkg/xds/bootstrap/template_v3.go:167]: Sanity check: is this also going to apply for ClusterLoadAssignment?

Comment (jijiechen) [reply pkg/xds/bootstrap/template_v3.go:167]: ClusterLoadAssignment is covered in EDS, located at `pkg/xds/envoy/clusters/v3/eds_cluster_configurer.go`

**Review by michaelbeaumont — COMMENTED** [review]
I think we need to set https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#envoy-v3-api-field-config-cluster-v3-cluster-eds-cluster-config

EDIT: right we're doing that

File: pkg/xds/bootstrap/template_v3.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/xds/bootstrap/template_v3.go:167]: Yeah we need to set that as well I think

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**